ComponentKitTests/TransactionalDataSource/CKTransactionalComponentDataSourceTests.mm

```diff
@@ -192,6 +196,34 @@ - (void)testSynchronousReloadCancelsPreviousAsynchronousReload
   }));
 }
 
+- (void)testAsynchronouslyInsertingItemsAppliesModificationsOnWorkThread
```

Comment:
I'm having a bit of trouble finding the best way to verify that the modifications are applied on the given work thread. Open to suggestions here.